src/posix/platform/mainloop.hpp

```diff
@@ -65,6 +65,12 @@ class Source
      */
     virtual void Process(const otSysMainloopContext &aContext) = 0;
 
+    /**
+     * This method marks desturctor virtual method.
+     *
+     */
+    virtual ~Source() = default;
```

Comment:
Can we define the destructor to be private so that subclasses can't override it?

Reply:
My thought on a destructor of a base class is it should be inheritable so that derived class can take advantage of it to do its own cleanup logic. If it's made private, I guess the derived will not be able to do so?

Reply:
Actually, this isn't a problem. And the compiler will raise error if there is a real misuse, i.e. delete subclass with base class pointer.

Reply:
> 

Agree on your point of having to link with stdc++. I will dig deeper on the dependency graph again why the lib is required.

Reply:
> 

But i thinks polymorphism is designed to allow operating on subclass with base class pointer.